Modules/_io/bufferedio.c

```diff
@@ -1767,6 +1778,27 @@ _bufferedwriter_reset_buf(buffered *self)
     self->write_end = -1;
 }
 
+static void
```

Comment:
This function should be checked for failure, so it should not return `void`.

Reply:
I'm not so sure.  What is its failure state?  It either sets `self->appending = 1` or `self->appending = 0`.  Neither case represents a failure or success.  AFAICT all calls within it check for failure or success and are handled appropriately.

Reply:
`getattr()` can raise an arbitrary exception, for example MemoryError or KeyboardInterrupt. They should not be cleared.

BTW, in 3.13 you can use `PyObject_GetOptionalAttr()` which automatically silences an AttributeError. It exists under other name as private API in older versions.

`PyUnicode_FindChar()` also can raise an exception, according to the specification, although it is less likely after getting rid of legacy strings.